lib/taskana-core/src/test/java/acceptance/task/CallbackStateAccTest.java

```diff
@@ -155,13 +158,82 @@ void testUpdateOfCallbackState()
+        assertEquals(CallbackState.CLAIMED, createdTask2.getCallbackState());
+
+        TaskImpl createdTask3 = createTask(taskService, CallbackState.CALLBACK_PROCESSING_COMPLETED);
+        assertEquals(CallbackState.CALLBACK_PROCESSING_COMPLETED, createdTask3.getCallbackState());
+
+        List<String> externalIds = new ArrayList<>(
+            Arrays.asList(createdTask1.getExternalId(), createdTask2.getExternalId(), createdTask3.getExternalId()));
+
+        //try to set CallbackState to NONE
+        BulkOperationResults<String, TaskanaException> bulkResult1 = taskService.setCallbackStateForTasks(externalIds,
+            CallbackState.NONE);
+
+        //It's never allowed to set CallbackState to NONE over public API
+        assertTrue(bulkResult1.containsErrors());
+        List<String> failedTaskIds = bulkResult1.getFailedIds();
+        assertTrue(failedTaskIds.size() == 3);
+        //add removed externalIds again
+        failedTaskIds.forEach(externalId -> externalIds.add(externalId));
+
+        //complete a task
+        createdTask3 = (TaskImpl) taskService.forceCompleteTask(createdTask3.getId());
+
+        //It's only allowed to set CallbackState to COMPLETE, if TaskState equals COMPLETE, therefore 2 tasks should not get updated
+        BulkOperationResults<String, TaskanaException> bulkResult2 = taskService.setCallbackStateForTasks(externalIds,
+            CallbackState.CALLBACK_PROCESSING_COMPLETED);
+        assertTrue(bulkResult2.containsErrors());
+        List<String> failedTaskIds2 = bulkResult2.getFailedIds();
+        assertTrue(failedTaskIds2.size() == 2);
+        failedTaskIds2.forEach(externalId -> externalIds.add(externalId));
+
+        //claim two tasks
+        createdTask1 = (TaskImpl) taskService.forceClaim(createdTask1.getId());
+        createdTask2 = (TaskImpl) taskService.forceClaim(createdTask2.getId());
+
+        //It's only allowed to claim a task if the TaskState equals CLAIMED and the CallbackState equals REQUIRED
+        //Therefore 2 tasks should not get updated
+        BulkOperationResults<String, TaskanaException> bulkResult3 = taskService.setCallbackStateForTasks(externalIds,
+            CallbackState.CLAIMED);
+        assertTrue(bulkResult1.containsErrors());
+        List<String> failedTaskIds3 = bulkResult3.getFailedIds();
+        assertTrue(failedTaskIds3.size() == 2);
+        failedTaskIds3.forEach(externalId -> externalIds.add(externalId));
+
+        //It's only allowed to set the CallbackState to REQUIRED if the TaskState doesn't equal COMPLETE
+        //Therefore 1 task should not get updated
+        BulkOperationResults<String, TaskanaException> bulkResult4 = taskService.setCallbackStateForTasks(externalIds,
+            CallbackState.CALLBACK_PROCESSING_REQUIRED);
+        assertTrue(bulkResult4.containsErrors());
+        List<String> failedTaskIds4 = bulkResult4.getFailedIds();
+        assertTrue(failedTaskIds4.size() == 1);
+
+    }
```

Comment:
Test to loooooonnngg ... can you split them up into multiple tests?

Reply:
Done